modules/axfr/axfr.go

```diff
@@ -148,10 +147,11 @@ func (s *GlobalLookupFactory) AddFlags(f *flag.FlagSet) {
 	f.StringVar(&s.BlacklistPath, "blacklist-file", "", "blacklist file for servers to exclude from AXFR lookups")
 }
 
-func (s *GlobalLookupFactory) MakeRoutineFactory() (zdns.RoutineLookupFactory, error) {
+func (s *GlobalLookupFactory) MakeRoutineFactory(threadID int) (zdns.RoutineLookupFactory, error) {
```

Comment:
This really seems like something that should be available as a global or something and not something that we have to be passing around. I remain _very_ skeptical about this.